relay-server/src/actors/project.rs

```diff
@@ -379,6 +382,42 @@ enum GetOrFetch<'a> {
     Scheduled(&'a mut StateChannel),
 }
 
+/// Represents either the project state or an aggregation of metrics.
+///
+/// We have to delay rate limiting on metrics until we have a valid project state,
+/// So when we don't have one yet, we hold them in this aggregator until the project state arrives.
+///
+/// TODO: spool queued metrics to disk when the in-memory aggregator becomes too full.
+#[derive(Debug)]
+enum State {
+    Cached(Arc<ProjectState>),
+    Pending(Box<aggregator::Aggregator>),
+}
+
+impl State {
+    fn state_value(&self) -> Option<Arc<ProjectState>> {
+        match self {
+            State::Cached(state) => Some(Arc::clone(state)),
+            State::Pending(_) => None,
+        }
+    }
+
+    /// Sets the new `ProjectState`. If the variant was pending, the buckets will be returned.
```

Comment:
This comment a bit misleading here. 
```suggestion
    /// Sets the cached state using provided `ProjectState`. 
    /// If the variant was pending, the buckets will be returned.
```